keps/sig-apps/20190318-PodDisruptionBudget-graduation-to-stable.md

```diff
@@ -80,6 +80,60 @@ modifications and use the updated values when making decisions.
 
 This feature is implemented by [this PR](https://github.com/kubernetes/kubernetes/pull/69867).
 
+#### Eviction of non-ready pods
+
+There are a couple of open issues where pods can't be evicted, even if they not Ready and Running 
+(https://github.com/kubernetes/kubernetes/issues/72320 and https://github.com/kubernetes/kubernetes/issues/80389).
+The root of this issue, is that the rules in the disruption controller for what is a healthy pod, and the rules
+in the Eviction API for when a pod can be evicted without looking at the PDB are not the same. This means a pod can
+be considered unhealthy by the disruption controller so it does not count as healthy when computing `DisruptionsAllowed`,
+but will still require `DisruptionsAllowed` to be larger than 0 for it to be evicted. Some strange situations can
+arise from this. For example if we have a PDB with `MinAvailable  = 1` and 10 pods that are all in the CrashLoop state 
+(`Running`, but not `Ready`), we will not be allowed to evict any of the pods.
+
+The proposal is that we fix this by making sure that all pods that are not considered healthy by the disruption
+controller can be evicted even if `DisruptionsAllowed` is 0. There is pretty much agreement that this is safe
+for pods in a phase other than `Running` and there is an open PR for this: https://github.com/kubernetes/kubernetes/pull/83906
+It should also be safe to evict `Running` pods that are not `Ready`. The `Ready` condition is already used
+by most of the workload controllers during rollouts and for determining which pods should receive traffic from
+a service. But there are some concerns that it might not be safe in all situations: 
+https://github.com/kubernetes/kubernetes/pull/83906#issuecomment-558287640
+
+#### Make the disruption controller more lenient for pods belonging to non-scale controllers
+
+The disruption controller is currently taking the very safe route whenever it encounters any
+issues with the targeted pods or their controllers. For all configurations of the PDB, except when
+`minAvailable` is a number, the PDB requires that it can find the controller and that
+the controller implements scale (either by being one of the built-in workloads or a CR where the CRD implements
+the scale subresource). If those conditions are not met for all pods targeted by the PDB, the disruption
+controller will set `DisruptionsAllowed` to 0, which means none of the pods can be evicted. There is an issue
+concerning this behavior: https://github.com/kubernetes/kubernetes/issues/77383.
+
+The current behavior of the disruption controller for the different types of input and the different
+types of pods that might be encountered are documented in: 
+https://docs.google.com/spreadsheets/d/12HUundBS-slA6axfQYZPRCeIu_Au_wsGD0Vu_oKAnM8/edit?usp=sharing
```

Comment:
once we finalize the content, we'll want to pull in the table to this doc rather than leave it referenced as an external spreadsheet

Reply:
Yeah I agree.